src/robot/libraries/dialogs_ipy.py

```diff
@@ -12,32 +12,183 @@
+        grid.RowDefinitions.Add(selection_row)
+        grid.RowDefinitions.Add(button_row)
+        self.Content = grid
 
-class InputDialog(_AbstractWinformsDialog):
+    def _create_body(self, message, value, **extra):
+        _label = Label()
+        textblock = TextBlock()
+        textblock.Text = message
+        textblock.TextWrapping = TextWrapping.Wrap
+        _label.Content = textblock
+        _label.Margin = Thickness(10)
+        _label.SetValue(Grid.ColumnSpanProperty, 2)
+        _label.SetValue(Grid.RowProperty, 0)
+        self.Content.AddChild(_label)
+        selector = self._create_selector(value, **extra)
+        if selector:
+            self.Content.AddChild(selector)
+            selector.Focus()
 
-    def __init__(self, message, default, hidden=False):
-        _AbstractWinformsDialog.__init__(self)
+    def _create_selector(self, value):
+        return None
 
+    def _create_buttons(self):
+        self.left_button = self._create_button(self._left_button, self._left_button_clicked)
+        self.left_button.SetValue(Grid.ColumnProperty, 0)
+        self.left_button.IsDefault = True
+        self.right_button = self._create_button(self._right_button, self._right_button_clicked)
+        if self.right_button:
+            self.right_button.SetValue(Grid.ColumnProperty, 1)
+            self.Content.AddChild(self.right_button)
+            self.right_button.IsCancel = True
+            self.left_button.SetValue(Grid.ColumnProperty, 0)
+            self.Content.AddChild(self.left_button)
+        else:
+            self.left_button.SetValue(Grid.ColumnSpanProperty, 2)
+            self.Content.AddChild(self.left_button)
 
-class SelectionDialog(_AbstractWinformsDialog):
+    def _create_button(self, content, callback):
+        if content:
+            button = Button()
+            button.Margin = Thickness(10)
+            button.MaxHeight = 50
+            button.MaxWidth = 150
+            button.SetValue(Grid.RowProperty, 2)
+            button.Content = content
+            button.Click += callback
+            return button
 
-    def __init__(self, message, options):
-        _AbstractWinformsDialog.__init__(self)
+    def _left_button_clicked(self, sender, eventArgs):
+        if self._validate_value():
+            self._result = self._get_value()
+            self._close()
 
+    def _validate_value(self):
+        return True
 
-class PassFailDialog(_AbstractWinformsDialog):
+    def _get_value(self):
+        return None
 
-    def __init__(self, message):
-        _AbstractWinformsDialog.__init__(self)
+    def _close(self):
+        self.Close()
+
+    def _right_button_clicked(self, sender, eventArgs):
+        self._result = self._get_right_button_value()
+        self._close()
+
+    def _get_right_button_value(self):
+        return None
+
+    def show(self):
+        if Application.Current == None:
+            parent = _mainWindow()
+        else:
+            parent = Application.MainWindow
```

Comment:
```parent``` is just a placeholder for the ```Application.MainWindow```. On the first 'atest', the application is launched with a minimized ```MainWindow```. On subsequent tests, the ```Application``` already exists and it's existing ```MainWindow``` is used... this was how I got around the AppDomain issue mentioned in the comments on ```class _mainWindow(Window):```